pkg/reconciler/apis/apibinding/apibinding_controller.go

```diff
@@ -301,9 +336,10 @@ func (c *controller) process(ctx context.Context, key string) error {
 	old := obj
 	obj = obj.DeepCopy()
 
-	if err := c.reconcile(ctx, obj); err != nil {
-		return err
-	}
+	reconcileErr := c.reconcile(ctx, obj)
+
+	// Regardless of whether reconcile returned an error or not, always try to patch status if needed. Return the
+	// reconciliation error at the end.
```

Comment:
(no action necessary) is this something we have to do elsewhere in other controllers too?